sound/soc/sof/intel/hda-stream.c

```diff
@@ -486,7 +486,7 @@ int hda_dsp_stream_hw_params(struct snd_sof_dev *sdev,
 	int sd_offset = SOF_STREAM_SD_OFFSET(hstream);
 	int ret;
 	u32 dma_start = SOF_HDA_SD_CTL_DMA_START;
-	u32 mask;
+	u32 mask = BIT(hstream->index);
```

Comment:
actually the initial code is rotten already. We assign hstream by dereferencing hext_stream BEFORE testing hext_stream. This is not right. both hstream and mask assignments need to be placed after the test for if (!hext_stream)

Reply:
Hah, that's true. I'll fix that as well with a separate patch.